src/Rules/FunctionDefinitionCheck.php

```diff
@@ -61,6 +62,23 @@ public function __construct(
 		$this->checkThisOnly = $checkThisOnly;
 	}
 
+	/**
+	 * @return string[]
+	 */
+	private function getSupportedTypehints(): array
+	{
+		if (PHP_VERSION_ID < 70200) {
+			$unsupportedTypehints = ['object'];
+			if (PHP_VERSION_ID < 70100) {
+				$unsupportedTypehints += ['void', 'iterable'];
+			}
+
+			return array_diff(self::VALID_TYPEHINTS, $unsupportedTypehints);
```

Comment:
I think that a reversed approach (define common ones in VALID_TYPEHINTS and add add the new ones on the method.) rather than unreadable `array_diff` call.